aiounifi/models/voucher.py

```diff
@@ -0,0 +1,229 @@
+"""Hotspot vouchers as part of a UniFi network."""
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import NotRequired, Self, TypedDict
+
+from .api import ApiItem, ApiRequest
+
+
+class TypedVoucher(TypedDict):
+    """Voucher type definition."""
+
+    _id: str
+    site_id: str
+    note: NotRequired[str]
+    code: str
+    quota: int
+    duration: float
+    qos_overwrite: NotRequired[bool]
+    qos_usage_quota: NotRequired[str]
+    qos_rate_max_up: NotRequired[int]
+    qos_rate_max_down: NotRequired[int]
+    used: int
+    create_time: float
+    start_time: NotRequired[float]
+    end_time: NotRequired[float]
+    for_hotspot: NotRequired[bool]
+    admin_name: str
+    status: str
+    status_expires: float
+
+
+@dataclass
+class VoucherListRequest(ApiRequest):
+    """Request object for voucher list."""
+
+    @classmethod
+    def create(cls) -> Self:
+        """Create voucher list request."""
+        return cls(
+            method="get",
+            path="/stat/voucher",
+        )
+
+
+@dataclass
+class VoucherCreateRequest(ApiRequest):
+    """Request object for voucher create."""
+
+    @classmethod
+    def create(
+        cls,
+        number: int,
+        quota: int,
+        expire_number: int,
+        expire_unit: int = 1,
+        usage_quota: int | None = None,
+        rate_max_up: int | None = None,
+        rate_max_down: int | None = None,
+        note: str | None = None,
+    ) -> Self:
+        """Create voucher create request.
+
+        :param number: number of vouchers
+        :param quota: number of using; 0 = unlimited
+        :param expire_number: expiration of voucher per expire_unit
+        :param expire_unit: scale of expire_number, 1 = minute, 60 = hour, 3600 = day
+        :param usage_quota: quantity of bytes allowed in MB
+        :param rate_max_up: up speed allowed in kbps
+        :param rate_max_down: down speed allowed in kbps
+        :param note: description
+        """
+        data = {
+            "cmd": "create-voucher",
+            "n": number,
+            "quota": quota,
+            "expire_number": expire_number,
+            "expire_unit": expire_unit,
+        }
+        if usage_quota:
+            data["bytes"] = usage_quota
+        if rate_max_up:
+            data["up"] = rate_max_up
+        if rate_max_down:
```

Comment:
```suggestion
        if rate_max_down is not None:
```
Unless 0 is not allowed then its ok

Reply:
0 is not allowed.